lib/grover/js/processor.cjs

```diff
@@ -2,7 +2,12 @@
 try {
   const Module = require('module');
   // resolve puppeteer from the CWD instead of where this script is located
-  var puppeteer = require(require.resolve('puppeteer', { paths: Module._nodeModulePaths(process.cwd()) }));
+  var puppeteer;
+  try {
+    puppeteer = require(require.resolve('puppeteer', { paths: Module._nodeModulePaths(process.cwd()) }));
+  } catch (e) {
+    puppeteer = require(require.resolve('puppeteer-core', { paths: Module._nodeModulePaths(process.cwd()) }));
```

Comment:
i'm not convinced this solves the problem without creating a new one. With your change the warning people will see if/when puppeteer isn't installed is that they need to install `puppeteer-core`, which is *not* the desired behaviour. Have you run the specs to test this? I have two specs testing exactly that and I don't see them passing with your change.

While on specs, this functionality is testable and I'd like to see any change include tests to show it works the way expected to prevent regressions.

Reply:
@abrom since `options.browserWsEndpoint` is needed to determine the proper dependency, can we move the `require`s either in `_processPage` or further down in the listener function?

Reply:
@abrom I made a new commit that passed all tests including the remote ones, I did have to change a lot more code though. let me know what you think.